File: Ent_UsAdmin/CrearNotificacionDescuentoPromesa.vb
Imports System.ComponentModel
Imports MySql.Data.MySqlClient

Public Class CrearNotificacionDescuentoPromesa
    Public idPromesa As String
    Dim dataUsuarios As DataTable
    Dim adapterUsuarios As MySqlDataAdapter
    Dim nCargando As Cargando
    Dim creada As Boolean

    Private Sub CrearNotificacionCancelarTicket_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        creada = False
        lblPromesa.Text = idPromesa
        nCargando = New Cargando
        nCargando.Show()
        nCargando.MonoFlat_Label1.Text = "Consultando usuarios"
        nCargando.TopMost = True
        BackgroundWorker1.RunWorkerAsync()


    End Sub

    Private Sub BackgroundWorker1_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork
        Dim conexionUsuarios As MySqlConnection
        conexionUsuarios = New MySqlConnection()
        conexionUsuarios.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS"
        conexionUsuarios.Open()


        Dim consulta As String


        consulta = "select Usr.nm,Usr.nm_complete from Usr inner join grp on Usr.grp = grp.id inner join PermisosGrupo on grp.id = PermisosGrupo.idGrupo where PermisosGrupo.SatiModCreditosDescuentoPromesa = 1"
        adapterUsuarios = New MySqlDataAdapter(consulta, conexionUsuarios)
        dataUsuarios = New DataTable
        adapterUsuarios.Fill(dataUsuarios)



    End Sub

    Private Sub BackgroundWorker1_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundWorker1.RunWorkerCompleted
        ComboUsuarios.Clear()
        For Each row As DataRow In dataUsuarios.Rows
            ComboUsuarios.AddItem(row("nm").ToString)
        Next
        ComboUsuarios.selectedIndex = 0
        nCargando.Close()
    End Sub

    Private Sub ComboUsuarios_onItemSelected(sender As Object, e As EventArgs) Handles ComboUsuarios.onItemSelected
        For Each row As DataRow In dataUsuarios.Rows
            If row("nm").ToString = ComboUsuarios.selectedValue Then
                lblNombreUsuario.Text = row("nm_complete").ToString
                Exit For
            End If
        Next

    End Sub

    Private Sub BackgroundNotificacion_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundNotificacion.DoWork
        Try
            Dim conexionLogin As MySqlConnection
            conexionLogin = New MySqlConnection()
            conexionLogin.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS"
            conexionLogin.Open()
            Dim comandoLogin As MySqlCommand
            Dim consultaLogin As String
            consultaLogin = "insert into Notificaciones values(null,'DescuentoPromesa',1,'" & nmusr & "','" & ComboUsuarios.selectedValue & "','" & idPromesa & "','','" & txtComentario.Text & "',0,'0','" & Date.Now.ToString("yyyy-MM-dd") & "','" & Date.Now.ToString("HH:mm:ss") & "',0,'','','" & nombreAmostrar & "','',''); "
            comandoLogin = New MySqlCommand
            comandoLogin.Connection = conexionLogin
            comandoLogin.CommandText = consultaLogin
            comandoLogin.ExecuteNonQuery()
            conexionLogin.Close()
            creada = True
        Catch ex As MySqlException
            creada = False
            MessageBox.Show(ex.Message)
        End Try

    End Sub

    Private Sub BunifuThinButton21_Click(sender As Object, e As EventArgs) Handles BunifuThinButton21.Click
        nCargando = New Cargando
        nCargando.Show()
        nCargando.MonoFlat_Label1.Text = "Notificando"
        nCargando.TopMost = True
        BunifuThinButton21.Enabled = False

        BackgroundNotificacion.RunWorkerAsync()


    End Sub

    Private Sub BackgroundNotificacion_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundNotificacion.RunWorkerCompleted
        If creada Then
            MessageBox.Show("Notificación creada con éxito")
            BunifuThinButton21.Enabled = True
            nCargando.Close()
            PromPago.creada = True
            Me.Close()
        Else
            MessageBox.Show("Ha ocurrido un error, inténtelo de nuevo")
            BunifuThinButton21.Enabled = True
            creada = False
            nCargando.Close()
        End If

    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub

    Private Sub CrearNotificacionDescuentoPromesa_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        If creada Then
            PromPago.creada = True
        Else
            PromPago.creada = False
        End If
    End Sub
End Class

File: Ent_UsAdmin/Notificaciones/CrearNotificacionDescuentoPromesa.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace ConfiaAdmin
{

    public partial class CrearNotificacionDescuentoPromesa
    {
        public string idPromesa;
        private DataTable dataUsuarios;
        private MySqlDataAdapter adapterUsuarios;
        private Cargando nCargando;
        private bool creada;
        public string tipoCredito;

        public CrearNotificacionDescuentoPromesa()
        {
            InitializeComponent();
        }

        private void CrearNotificacionCancelarTicket_Load(object sender, EventArgs e)
        {
            creada = false;
            lblPromesa.Text = idPromesa;
            nCargando = new Cargando();
            nCargando.Show();
            nCargando.MonoFlat_Label1.Text = "Consultando usuarios";
            nCargando.TopMost = true;
            BackgroundWorker1.RunWorkerAsync();


        }

        private void BackgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            MySqlConnection conexionUsuarios;
            conexionUsuarios = new MySqlConnection();
            conexionUsuarios.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS";
            conexionUsuarios.Open();


            string consulta;


            consulta = "select Usr.nm,Usr.nm_complete from Usr inner join grp on Usr.grp = grp.id inner join PermisosGrupo on grp.id = PermisosGrupo.idGrupo where PermisosGrupo.SatiModCreditosDescuentoPromesa = 1";
            adapterUsuarios = new MySqlDataAdapter(consulta, conexionUsuarios);
            dataUsuarios = new DataTable();
            adapterUsuarios.Fill(dataUsuarios);



        }

        private void BackgroundWorker1_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ComboUsuarios.Clear();
            foreach (DataRow row in dataUsuarios.Rows)
                ComboUsuarios.AddItem(row["nm"].ToString());
            ComboUsuarios.selectedIndex = 0;
            nCargando.Close();
        }

        private void ComboUsuarios_onItemSelected(object sender, EventArgs e)
        {
            foreach (DataRow row in dataUsuarios.Rows)
            {
                if ((row["nm"].ToString() ?? "") == (ComboUsuarios.selectedValue ?? ""))
                {
                    lblNombreUsuario.Text = row["nm_complete"].ToString();
                    break;
                }
            }

        }

        private void BackgroundNotificacion_DoWork(object sender, DoWorkEventArgs e)
        {
            try
            {
                MySqlConnection conexionLogin;
                conexionLogin = new MySqlConnection();
                conexionLogin.ConnectionString = "server=www.prestamosconfia.com;user id=ajas;pwd=123456;port=3306;database=USRS";
                conexionLogin.Open();
                MySqlCommand comandoLogin;
                string consultaLogin;
                consultaLogin = "insert into Notificaciones values(null,'DescuentoPromesa',1,'" + Module1.nmusr + "','" + ComboUsuarios.selectedValue + "','" + idPromesa + "','','" + txtComentario.Text + "',0,'0','" + DateTime.Now.ToString("yyyy-MM-dd") + "','" + DateTime.Now.ToString("HH:mm:ss") + "',0,'','','" + Module1.nombreAmostrar + "','',''); ";
                comandoLogin = new MySqlCommand();
                comandoLogin.Connection = conexionLogin;
                comandoLogin.CommandText = consultaLogin;
                comandoLogin.ExecuteNonQuery();
                conexionLogin.Close();
                creada = true;
            }
            catch (MySqlException ex)
            {
                creada = false;
                MessageBox.Show(ex.Message);
            }

        }

        private void BunifuThinButton21_Click(object sender, EventArgs e)
        {
            nCargando = new Cargando();
            nCargando.Show();
            nCargando.MonoFlat_Label1.Text = "Notificando";
            nCargando.TopMost = true;
            BunifuThinButton21.Enabled = false;

            BackgroundNotificacion.RunWorkerAsync();


        }

        private void BackgroundNotificacion_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            if (creada)
            {
                MessageBox.Show("Notificación creada con éxito");
                BunifuThinButton21.Enabled = true;
                nCargando.Close();
                My.MyProject.Forms.PromPago.creada = true;
                Close();
            }
            else
            {
                MessageBox.Show("Ha ocurrido un error, inténtelo de nuevo");
                BunifuThinButton21.Enabled = true;
                creada = false;
                nCargando.Close();
            }

        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }

        private void CrearNotificacionDescuentoPromesa_FormClosing(object sender, FormClosingEventArgs e)
        {
            if (creada)
            {
                if (tipoCredito == "L")
                {
                    My.MyProject.Forms.PromPagoLegal.creada = true;
                }
                else
                {
                    My.MyProject.Forms.PromPago.creada = true;
                }
            }

            else if (tipoCredito == "L")
            {
                My.MyProject.Forms.PromPagoLegal.creada = false;
            }
            else
            {
                My.MyProject.Forms.PromPago.creada = false;

            }
        }
    }
}